### PR TITLE
Setup docs app

### PR DESCRIPTION
**Pull request by simonihmig**
This is only the boilerplate of the docs app, copied from ember-toucan-core, and enabling CI/CD. Writing the actual documentation will follow in separate PR(s).

### CODE REVIEW

**Comment by simonihmig** [comment]
@NullVoxPopuli [this message](https://github.com/CrowdStrike/ember-headless-form/actions/runs/4165946401/jobs/7209639446#step:6:13) seems to suggest we need to create a new project for this? Do you have the permissions, I don't have any cloudflare access on behalf of CS, right?

**Comment by NullVoxPopuli** [comment]
> @NullVoxPopuli [this message](https://github.com/CrowdStrike/ember-headless-form/actions/runs/4165946401/jobs/7209639446#step:6:13) seems to suggest we need to create a new project for this? 

I _just_ made it 🎉 

> I don't have any cloudflare access on behalf of CS, right?

Right, we can get you added if you want (UX has its own cloudflare account). But for this kind of setup you _also_ would need to be a GitHub org admin. :(

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -167,62 +167,49 @@ jobs:
...
+    outputs:
+      url: ${{ steps.publishStep.outputs.url }}
+      env: ${{ steps.publishStep.outputs.environment }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+      - uses: ./.github/actions/pnpm
+      - run: pnpm build
+      # - uses: ./.github/actions/download-built-package
```
Comment (ynotdraw) [inline .github/workflows/ci.yml:186]: Can this be removed since we have the build `run` above+below?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -167,62 +167,49 @@ jobs:
...
+    name: Post Preview URL as comment to PR
+    runs-on: ubuntu-latest
+    needs: PublishDocstoCloudflarePages
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+      - uses: ./.github/actions/pnpm
+      - run: pnpm build
+      # - uses: ./.github/actions/download-built-package
```
Comment (ynotdraw) [inline .github/workflows/ci.yml:208]: Same question here as above

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -167,62 +167,49 @@ jobs:
...
+      - uses: marocchino/sticky-pull-request-comment@v2
+        with:
+          message: |+
+            ## Preview URLs
+            Env: ${{ needs.PublishDocstoCloudflarePages.outputs.env }}
+            Docs: ${{ needs.PublishDocstoCloudflarePages.outputs.url }}
+#            # api docs: ${{ needs.PublishDocstoCloudflarePages.outputs.url }}/api/modules.html
```
Comment (ynotdraw) [inline .github/workflows/ci.yml:215]: I should maybe do this as well in toucan-core, as we don't have api docs yet!

File: docs-app/app/components/feature-card.gts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    <h4 class="pb-4 type-2xl text-titles-and-attributes">
+      {{@title}}
+    </h4>
+    <p class="text-body-and-labels">
+      {{yield}}
+    </p>
+  </div>
+</template>;
+
+export default FeatureCard;
```
Comment (ynotdraw) [inline docs-app/app/components/feature-card.gts:20]: Maybe we can/should move these over to `oss-docs` eventually? (Not part of this PR, obvs, low priority!)

File: docs-app/app/index.html [hunk 5]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <title>ember-headless-form</title>
+    <meta name="description" content="components in ember-toucan-core" />
```
Comment (ynotdraw) [inline docs-app/app/index.html:6]:
```suggestion
    <meta name="description" content="components in ember-headless-form" />
```

File: docs-app/ember-cli-build.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const EmberApp = require('ember-cli/lib/broccoli/ember-app');
+
+const isProduction = () => EmberApp.env() === 'production';
+
+module.exports = function (defaults) {
+  let app = new EmberApp(defaults, {
+    autoImport: {
+      watchDependencies: ['@crowdstrike/ember-toucan-core'],
```
Comment (ynotdraw) [inline docs-app/ember-cli-build.js:10]:
```suggestion
      watchDependencies: ['ember-headless-form'],
```

File: docs-app/package.json [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    "@embroider/router": "^2.0.0",
+    "ember-browser-services": "^4.0.4",
+    "ember-cached-decorator-polyfill": "^1.0.1",
+    "ember-headless-form": "workspace:*",
+    "ember-modifier": "^4.0.0",
+    "ember-resources": "^5.4.0",
+    "highlight.js": "^11.6.0",
+    "highlightjs-glimmer": "^1.4.1",
+    "tracked-built-ins": "^3.1.0"
+  }
```
Comment (ynotdraw) [inline docs-app/package.json:128]: Do you need the `dependenciesMeta` here as well? E.g.,:

```
"dependenciesMeta": {
    "ember-headless-form": {
      "injected": true
    }
  }
```

(https://github.com/CrowdStrike/ember-toucan-core/blob/main/docs-app/package.json#L134)

File: docs-app/tests/helpers/index.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  // Additional setup for rendering tests can be done here.
+}
+
+function setupTest(hooks, options) {
+  upstreamSetupTest(hooks, options);
+
+  // Additional setup for unit tests can be done here.
+}
+
+export { setupApplicationTest, setupRenderingTest, setupTest };
```
Comment (ynotdraw) [inline docs-app/tests/helpers/index.js:42]: Totally non-blocking quesiton: We have all of the same in `toucan-core` under `docs-app/tests`, but I'm curious if it makes sense for our docs-apps to even have the `tests/` directory?  We normally use the `test-app` for testing, so just curious on thoughts around this.

File: docs-app/.prettierrc.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+module.exports = {
+  plugins: ['prettier-plugin-ember-template-tag'],
```
Comment (nicolechung) [inline docs-app/.prettierrc.js:4]: !! Should we be using this in our other packages? Does it format everything in between `<template>` tags (is this what it does?)

File: docs-app/README.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+## Installation
+
+- `git clone <repository-url>` this repository
+- `cd docs-app`
+- `npm install`
+
+## Running / Development
+
+- `ember serve`
```
Comment (nicolechung) [inline docs-app/README.md:23]: Question...are we really running `ember serve` or is it `pnpm start`? Or should updating this `README` be in a followup?

File: docs-app/package.json (hunk 7 above)
Comment (nicolechung) [reply docs-app/package.json:128]: It seems like it was working without it based on what I read from a previous PR (let me dig it up)...but then how come we need it in other packages?? (don't feel the need to answer this if it's going to take a lot of investigation btw, just curious)

Comment (nicolechung) [reply docs-app/package.json:128]: Update, yeah you are right it was added over here: https://github.com/CrowdStrike/ember-headless-form/pull/49/files#diff-b03f4acb30e04ceb12a40763ef649759b3f632df2357bcec14fab6dd29ebbfb4R124 in the test app

File: docs-app/tests/helpers/index.js (hunk 8 above)
Comment (NullVoxPopuli) [reply docs-app/tests/helpers/index.js:42]: over time docs may have demos complex enough where you want them tested

File: docs-app/README.md (hunk 10 above)
Comment (NullVoxPopuli) [reply docs-app/README.md:23]: they are equiv 🎉

File: docs-app/tests/helpers/index.js (hunk 8 above)
Comment (ynotdraw) [reply docs-app/tests/helpers/index.js:42]: Ah, that makes sense 👍

File: .github/workflows/ci.yml (hunk 1 above)
Comment (simonihmig) [reply .github/workflows/ci.yml:186]: The other jobs also still have this. We should clean up all of this of course, when we have the final solution, which I hope #52 will bring!

File: docs-app/app/components/feature-card.gts (hunk 4 above)
Comment (simonihmig) [reply docs-app/app/components/feature-card.gts:20]: Yeah, I think this was even mentioned somewhere, in a comment or so. Just vaguely remember...

File: docs-app/.prettierrc.js (hunk 9 above)
Comment (simonihmig) [reply docs-app/.prettierrc.js:4]: We have this already in the test-app for example. It makes prettier even understand the existence of `<template>`, otherwise it wouldn't be able to parse the file (as it's not valid JS/TS). And yes, it does also format everything inside of the template!

File: docs-app/README.md (hunk 10 above)
Comment (simonihmig) [reply docs-app/README.md:23]: This was copy-paste, and yes, currently there is no difference. However, should we start to add turbo also to scripts in the workspace, including `start`, then it would start to matter!

File: docs-app/tests/helpers/index.js (hunk 8 above)
Comment (simonihmig) [reply docs-app/tests/helpers/index.js:42]: Yeah, and I would also like to add even just a very simple smoke test for the docs app, mainly checking that the build does not blow up, and at least something is rendered. This has proven to be useful in other projects like ember-bootstrap, to make sure that e.g. automated renovate updates don't accidentally the docs app!

File: docs-app/app/index.html (hunk 5 above)
Comment (simonihmig) [reply docs-app/app/index.html]: Thanks, applied!

File: docs-app/package.json (hunk 7 above)
Comment (simonihmig) [reply docs-app/package.json:133]: It wasn't needed as I wasn't really using the addon in the docs app yet. But yes, your are right, we should do that here also - and I just did that, to prevent any more foot guns due to this unfortunate issue.

**Comment by simonihmig** [comment]
@NullVoxPopuli I took the freedom to refactor the docs CD workflow a bit, in this [commit](https://github.com/CrowdStrike/ember-headless-form/pull/44/commits/c324e7aa249a8eb6b764bbae37dbc84e92131d37). Previously we had a separate `PostPreviewURLascommenttoPR` job, which seemed too much to me tbh. It did a `pnpm i` and `pnpm build`, which seems unnecessary for the `sticky-pull-request-comment` action. And if you remove those, what is left then is a single step that takes 1s, which IMO doesn't justify a separate job. So I merged the two together.

Btw, that `sticky-pull-request-comment` action wouldn't really be needed, if coudflare would merge and release [this bugfix](https://github.com/cloudflare/pages-action/pull/56), as the cloudflare action will create a `deployment` Github event that makes GH render it as a deployment in its native UI (which due to the bug doesn't work correctly for PRs rn). We could still leave the automated comment in here if we think that's useful though.

**Review by ynotdraw — APPROVED** [review]
